app/src/main/java/com/nervousfish/nervousfish/modules/pairing/AndroidBluetoothHandler.java

```diff
@@ -288,7 +288,7 @@ public void run() {
             BluetoothSocket socket;
 
             // Listen to the server socket if we're not connected
-            while (mState != AndroidBluetoothHandler.State.connected) {
+            while (mState != AndroidBluetoothHandler.State.connected && serverSocket != null) {
```

Comment:
This prevents a crash for when the serverSocket is null. Not really sure what it does though, maybe you could explain that